Rubberduck.Core/AutoComplete/SelfClosingPairCompletion/SelfClosingPairCompletionService.cs

```diff
@@ -61,11 +61,16 @@ private CodeString HandleOpeningChar(SelfClosingPair pair, CodeString original)
         {
             var nextPosition = original.CaretPosition.ShiftRight();
             var autoCode = new string(new[] { pair.OpeningChar, pair.ClosingChar });
-            var lines = original.Code.Split('\n');
+            var lines = original.Lines;
             var line = lines[original.CaretPosition.StartLine];
-            lines[original.CaretPosition.StartLine] = line.Insert(original.CaretPosition.StartColumn, autoCode);
-
-            return new CodeString(string.Join("\n", lines), nextPosition, original.SnippetPosition);
+            lines[original.CaretPosition.StartLine] = 
+                string.IsNullOrEmpty(original.Code) 
+                    ? autoCode 
+                    : original.CaretPosition.StartColumn == line.Length 
+                        ? line + autoCode 
+                        : line.Insert(original.CaretPosition.StartColumn, autoCode);
+
+            return new CodeString(string.Join("\r\n", lines), nextPosition, original.SnippetPosition);
         }
 
         private CodeString HandleClosingChar(SelfClosingPair pair, CodeString original)
```

Comment:
Why does a CodeString deal with closing pairs? That seems like an overreach..

Reply:
This is the SCP service... I don't understand this comment :confused:

Reply:
`HandleOpeningChar` handles opening-character inputs, `HandleClosingChar` handles closing-character inputs; both are necessary and in-scope here IMO.